src/Microsoft.OData.Edm/Validation/EdmError.cs

```diff
@@ -21,17 +21,35 @@ public class EdmError
         /// <param name="errorCode">An integer code representing the error.</param>
         /// <param name="errorMessage">A human readable message describing the error.</param>
         public EdmError(EdmLocation errorLocation, EdmErrorCode errorCode, string errorMessage)
+         : this(errorLocation, errorCode, errorMessage, EdmErrorLevel.Undefined)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the EdmError class.
+        /// </summary>
+        /// <param name="errorLocation">The location where the error occurred.</param>
+        /// <param name="errorCode">An integer code representing the error.</param>
+        /// <param name="errorMessage">A human readable message describing the error.</param>
+        /// <param name="errorLevel">An enum value describing the error severity.</param>
+        public EdmError(EdmLocation errorLocation, EdmErrorCode errorCode, string errorMessage, EdmErrorLevel errorLevel)
         {
             this.ErrorLocation = errorLocation;
             this.ErrorCode = errorCode;
             this.ErrorMessage = errorMessage;
+            this.ErrorLevel = errorLevel;
         }
 
         /// <summary>
         /// Gets the location of the error in the file in which it occurred.
         /// </summary>
         public EdmLocation ErrorLocation { get; private set; }
 
+        /// <summary>
+        /// Gets a string describing the error severity.
+        /// </summary>
+        public EdmErrorLevel ErrorLevel { get; private set; }

```

Comment:
Mike's PR for URL validation is adding another enum for similar purpose. Please sync with him to keep either ErrorLevel or MessageSeverity.